src/orcania.c

```diff
@@ -107,7 +116,7 @@ char * o_strdup(const char * source) {
 char * o_strndup(const char * source, size_t len) {
   char *new_str;
 
-  if (source == NULL || len < 0) {
+  if (source == NULL || (int)len < 0) {
```

Comment:
not necessary to cast len, a `size_t` variable can be tested with 0

Reply:
Could you compile it with `-Wall -Werror -Wextra`?

Reply:
OK, compiling with `-Wall -Werror -Wextra` tells me:
```
orcania.c:110:29: error: comparison of unsigned expression < 0 is always false [-Werror=type-limits]
   if (source == NULL || len < 0) {
```

Which means the comparison is not necessary at all, this line should be then:

```C
if (source == NULL) {
```

Reply:
Fixed https://github.com/silvioprog/orcania/commit/62e3258b1dca62d49be8a70c9a648e122fa1514b .